src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -3302,7 +3229,7 @@ public double getDroppableTombstoneRatio()
     @Override
     public long trueSnapshotsSize()
     {
-        return getDirectories().trueSnapshotsSize();
+        return SnapshotManager.instance.trueSnapshotsSize(metadata(), directories);
```

Comment:
NIT: we are using `getDirectories()` elsewhere
```suggestion
        return SnapshotManager.instance.trueSnapshotsSize(metadata(), getDirectories());
```

Reply:
I'm trying this out, and it is returning true size of 0 for me. We build a `SSTableSizeSummer` like this:

```
        SSTableSizeSummer visitor = new SSTableSizeSummer(metadata, directories
                                                                    .sstableLister(Directories.OnTxnErr.THROW)
                                                                    .listFiles());
```

the second parameter is a list of files to skip. Basically we end up skipping every sstable component file from the summer (adder is a better name?) . And we end up with no files to consider for the true size. Is this behavior desired/correct?

Maybe some unit tests are missing here to validate the expected behavior at the CFS level?

Reply:
I ll get back to you with the feedback, not sure what is going on here yet as I look at this.

Reply:
@frankgh 

Yes, this is intended behavior. The way it works is that there are two "sizes" of a snapshot. "true allocated size" of a snapshot in a snapshot directory is the size of all files which are located in the snapshot dir BUT THEY ARE NOT IN HOT DATA DIR.

In other words, because a snapshot is just a collection of hardlinks, then if you take a snapshot then its size is ... 0. Why? Because every file in a snapshot has its counterpart in a hot / live data dir.

But when some SSTables in hot data dir are compacted away, for example, then the hardlink / file in snapshot dir suddenly becomes "real" and it starts to be counted towards true snapshot size ...

Reply:
makes sense now, thanks for the clarification.